File: src/Software Engineer Prepare Kit/07/solution.ts
export default function maximizeNonOverlappingMeetings(meetings: number[][]): number {
	meetings.sort((a, b) => {
		return a[1]! - b[1]!
	})

	let maxNonOverlappingMeetings = 1

	for (let i = 0; i < meetings.length; i++) {
		let nonOverlappingMeetings = 1

		for (let j = i + 1; j < meetings.length; j++) {
			if (meetings[i]![1]! <= meetings[j]![0]!) {
				meetings[i]![1]! = meetings[j]![1]!
				nonOverlappingMeetings += 1
			}
		}

		maxNonOverlappingMeetings = Math.max(nonOverlappingMeetings, maxNonOverlappingMeetings)
	}

	return maxNonOverlappingMeetings
}

File: src/Software Engineer Prepare Kit/07/solution.go
package main

import "sort"

func MaximizeNonOverlappingMeetings(meetings [][]int32) int32 {
	sort.Slice(meetings, func(i, j int) bool {
		return meetings[i][1] < meetings[j][1]
	})

	var maxNonOverlappingMeetings int32 = 1

	for i := range meetings {
		var nonOverlappingMeetings int32 = 1

		for j := i + 1; j < len(meetings); j++ {
			if meetings[i][1] <= meetings[j][0] {
				meetings[i][1] = meetings[j][1]
				nonOverlappingMeetings += 1
			}
		}

		maxNonOverlappingMeetings = max(
			nonOverlappingMeetings,
			maxNonOverlappingMeetings,
		)
	}

	return maxNonOverlappingMeetings
}
